src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -535,38 +535,40 @@ impl<S: StateStore> SyncedKvLogStoreExecutor<S> {
         truncation_offset: Option<ReaderTruncationOffsetType>,
         buffer: &mut SyncedLogStoreBuffer,
     ) -> StreamExecutorResult<LogStorePostSealCurrentEpoch<'a, S::Local>> {
-        let epoch = barrier.epoch.prev;
-        let mut writer = write_state.start_writer(false);
-        writer.write_barrier(epoch, barrier.is_checkpoint())?;
+        // TODO(kwannoel): As an optimization we can also change flushed chunks to be flushed items
+        // to reduce memory consumption of logstore.
 
-        // FIXME(kwannoel): Flush all unflushed chunks
-        // As an optimization we can also change it into flushed items instead.
-        // This will reduce memory consumption of logstore.
+        let epoch = barrier.epoch.prev;
+        if barrier.is_checkpoint() {
+            let mut writer = write_state.start_writer(false);
+            writer.write_barrier(epoch, barrier.is_checkpoint())?;
```

Comment:
We may need to start the writer and write the barrier anyway instead of only on checkpoint barrier. Otherwise, we may lose the non-checkpoint barrier.

Reply:
Fixed